### PR TITLE
test: add prettier e2e test

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Tests Added + Pass?      | Yes
| License                  | MIT

Added prettier e2e test, which takes around 4mins on GitHub workflows.

For CI performance, only `jsx` and `js` format tests are executed. 

@fisker Please let me know if I miss any other tests related to the prettier-parser-babel.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14058"></a>

### CODE REVIEW

**Comment by fisker** [comment]
Some files under `tests/format/misc` also related to babel.

File: scripts/integration-tests/e2e-prettier.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+startLocalRegistry "$root"/verdaccio-config.yml
+yarn install
+
+if [ "$BABEL_8_BREAKING" = true ] ; then
+  # This option is removed in Babel 8
+  sed -i 's/allowDeclareFields: true,\?/\/* allowDeclareFields: true *\//g' babel.config.js
+fi
+
+# Only run js and jsx format tests
+yarn test "tests/format/jsx?"
```
Comment (JLHwung) [inline scripts/integration-tests/e2e-prettier.sh:48]: Should we add `--update-snapshot` here? I will expect the snapshot need updated when an error message is modified. Will `--update-snapshot` fail when a valid program throws or versa?

Comment (fisker) [reply scripts/integration-tests/e2e-prettier.sh:48]: I'm afraid  `--update-snapshot` can't be used here. Let me see if we can add an env flag to skip snapshot error messages, but only assert error throws.

Comment (fisker) [reply scripts/integration-tests/e2e-prettier.sh:48]: I tried to use different assertion, but it doesn't work. I guess we'll have to use `--update-snapshot`.

@sosukesuzuki WDYT?

Comment (sosukesuzuki) [reply scripts/integration-tests/e2e-prettier.sh:48]: I think we have to use `--update-snapshot`.

File: scripts/integration-tests/e2e-prettier.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+source utils/local-registry.sh
+source utils/cleanup.sh
+
+# Echo every command being executed
+set -x
+
+# Clone prettier
+git clone --depth=1 https://github.com/prettier/prettier /tmp/prettier
+cd /tmp/prettier || exit
```
Comment (nicolo-ribaudo) [inline scripts/integration-tests/e2e-prettier.sh:19]:
```suggestion
git clone --depth=1 https://github.com/prettier/prettier tmp/prettier
cd tmp/prettier || exit
```

Just to match the other e2e tests (it's a .gitignored folder)

**Review by nicolo-ribaudo — COMMENTED** [review]
Can we also run it on CircleCI for Babel 8?

File: scripts/integration-tests/e2e-prettier.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+startLocalRegistry "$root"/verdaccio-config.yml
+yarn install
+
+if [ "$BABEL_8_BREAKING" = true ] ; then
+  # This option is removed in Babel 8
+  sed -i 's/allowDeclareFields: true,\?/\/* allowDeclareFields: true *\//g' babel.config.js
+fi
+
+# Only run js,jsx,misc format tests
+yarn test "tests/format/(jsx?|misc)" --update-snapshot
```
Comment (fisker) [inline scripts/integration-tests/e2e-prettier.sh:48]:
```suggestion
yarn test "tests/format/(jsx?|misc)/" --update-snapshot
```

**Comment by JLHwung** [comment]
The prettier e2e test times out on Circle CI (>10min) while it takes 5min on GitHub workflows.

**Comment by nicolo-ribaudo** [comment]
Could you check if `--runInBand` makes it work on Circle?

**Comment by JLHwung** [comment]
Babel 8 test on `tests/format/jsx/fbt/jsfmt.spec.js` seems hanging, I will investigate.

**Comment by nicolo-ribaudo** [comment]
We can merge the Babel 7 tests for now, while you investigate.

**Comment by Yoyoqw** [comment]
> Q                      	A
> Tests Added + Pass?	Yes
> License	MIT
> Added prettier e2e test, which takes around 4mins on GitHub workflows.
> 
> For CI performance, only `jsx` and `js` format tests are executed.
> 
> @fisker Please let me know if I miss any other tests related to the prettier-parser-babel.
> 
> [](https://gitpod.io/#https://github.com/babel/babel/pull/14058)

Get off

**Comment by JLHwung** [comment]
I can not reproduce the prettier-e2e test error against Babel 8, they are all passed on my local environment. I assume Circle CI fails due to time out.

**Comment by nicolo-ribaudo** [comment]
CircleCI fails because there is no test output for 10 mins, there is probably something stuck :thinking:

**Comment by fisker** [comment]
Does it work without `BABEL_8_BREAKING=true`?

**Comment by nicolo-ribaudo** [comment]
Yes! https://github.com/babel/babel/runs/4732491820

@JLHwung If prettier uses jest, maybe we can try with `--runInBand`

**Comment by nicolo-ribaudo** [comment]
Similar problem: https://github.com/babel/babel/pull/13618

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**